### PR TITLE
Deprecate macros.getType and improve documentation in related procs.

### PR DESCRIPTION
**Pull request by dom96**
Refs #3681.

### CODE REVIEW

**Comment by dom96** [comment]
I thought that ``getTypeImpl`` and ``getType`` were the same, but that doesn't seem to be the case.

**Comment by krux02** [comment]
@dom96 to my understanding, ``getTypeImpl`` and ``getTypeInst`` are complete, and there is no need for another ``getType``. I am not sure though if all cases of the old ``getType`` are exactly the same as ``getTypeImpl``, but I am pretty sure that all use cases should be covered by the use of the combination of the two new functions. In other words, I support this pull request.

File: lib/core/macros.nim [hunk 1]
```diff
@@ -179,35 +179,68 @@ proc floatVal*(n: NimNode): BiggestFloat {.magic: "NFloatVal", noSideEffect.}
...
 proc getTypeImpl*(n: NimNode): NimNode {.magic: "NGetType", noSideEffect.}
-  ## Like getType except it includes generic parameters for the implementation
+  ## Returns the node's `type`:idx:.
+  ##
+  ## A Nim type is
+  ## mapped to a Nim AST too, so it's slightly confusing but it means the same
+  ## API can be used to traverse types. Recursive types are flattened for you
+  ## so there is no danger of infinite recursions during traversal.
+  ##
+  ## To resolve recursive types, you have to call 'getType' again.
```
Comment (yglukhov) [inline lib/core/macros.nim]: Did you really mean `getType` here? Thats a bit confusing, because its deprecated :)

Comment (dom96) [inline lib/core/macros.nim]: I just copied it from the older docs, so yeah, that's incorrect. I'm not sure whether this PR is correct any more at all though.

**Comment by Araq** [comment]
``getType`` is hard to deprecate, the required changes are not obivous and require a decent analysis by the programmer.

**Comment by krux02** [comment]
Well I think new code should not use ``getType`` anymore.

**Comment by johnnovak** [comment]
I still think it would be good to deprecate `getType` to discourage it's use. Then you might never actually remove it, or maybe in version 2 or something.

I'm just messing around with types in macros for the first time, and `getType` definitely sent me down the wrong path. It's hard to figure out that you have to use in fact `getTypeInst` if you want to get the name of the type as well (which you usually want).

**Pull request closed without merging**